graph.c

```diff
@@ -119,7 +119,7 @@ nodepath(struct node *n, bool escape)
 	escape = false;
 	nquote = 0;
 	for (s = n->path->s; *s; ++s) {
-		if (!isalnum(*s) && !strchr("_+-./", *s))
+		if (!isalnum((int)(*s)) && !strchr("_+-./", *s))
```

Comment:
The warning is legitimate, but this is not a correct fix. The C standard says that if the value is not representable as `unsigned char`, we get undefined behavior. If `char` is signed, and has a negative value (for example, one of the code units of some multibyte UTF-8 sequence), then converting to `int` still retains the negative value.

Instead, we want to interpret the the pointed-to `char` as `unsigned char`:

```C
isalnum(*(unsigned char *)s);
```